packages/playwright/src/index.ts

```diff
@@ -322,6 +322,8 @@ const playwrightFixtures: Fixtures<TestFixtures, WorkerFixtures> = ({
     clientInstrumentation.removeListener(csiListener);
     await artifactsRecorder.didFinishTest();
 
+    if (process.env.PLAYWRIGHT_COPY_PROMPT && (testInfo as TestInfoImpl)._isFailure())
```

Comment:
You can compute it from public fields, no need to violate layers. I'd rather we fixed other places too.